test/Sema/access-level-import-inlinable.swift

```diff
@@ -0,0 +1,159 @@
+// Test use of decls restricted by an import access-level in inlinable code.
+
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+/// Build the libraries.
+// RUN: %target-swift-frontend -emit-module %t/PublicLib.swift -o %t \
+// RUN:   -enable-library-evolution
+// RUN: %target-swift-frontend -emit-module %t/PackageLib.swift -o %t \
+// RUN:   -enable-library-evolution
+// RUN: %target-swift-frontend -emit-module %t/InternalLib.swift -o %t \
+// RUN:   -enable-library-evolution
+// RUN: %target-swift-frontend -emit-module %t/FileprivateLib.swift -o %t \
+// RUN:   -enable-library-evolution
+// RUN: %target-swift-frontend -emit-module %t/PrivateLib.swift -o %t \
+// RUN:   -enable-library-evolution
+
+/// Check diagnostics.
+// RUN: %target-swift-frontend -typecheck %t/Client.swift -I %t \
+// RUN:   -enable-library-evolution -swift-version 5 \
+// RUN:   -enable-experimental-feature AccessLevelOnImport -verify
+
+//--- PublicLib.swift
+public protocol PublicImportProto {
+    associatedtype T
+}
+public struct PublicImportType {
+    public init() {}
+}
+public func PublicFunc() {}
+
+@propertyWrapper
+public struct PublicImportWrapper<T> {
+  public var wrappedValue: T
+  public init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+}
+
+//--- PackageLib.swift
+public struct PackageImportType {
+    public init() {}
+}
+
+//--- InternalLib.swift
+public protocol InternalImportProto {
+    associatedtype T
+}
+public struct InternalImportType {
+    public init() {}
+}
+public func InternalFunc() {}
+
+//--- FileprivateLib.swift
+public protocol FileprivateImportProto {
+    associatedtype T
+}
+
+@propertyWrapper
+public struct FileprivateImportWrapper<T> {
+  public var wrappedValue: T
+  public init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+}
+
+//--- PrivateLib.swift
+public struct PrivateImportType {
+    public init() {}
+}
+
+//--- Client.swift
+public import PublicLib
+package import PackageLib // expected-note 2 {{module 'PackageLib' imported as 'package' here}}
+internal import InternalLib // expected-note 12 {{module 'InternalLib' imported as 'internal' here}}
+fileprivate import FileprivateLib // expected-note 6 {{module 'FileprivateLib' imported as 'fileprivate' here}}
+private import PrivateLib // expected-note 6 {{module 'PrivateLib' imported as 'private' here}}
```

Comment:
I'm thinking about adding the name of the affected decl to this diagnostic. While the current version works well in the terminal it may be harder to follow in an IDE when a big file is open.